imap_processing/swe/l1b/swe_l1b_science.py

```diff
@@ -0,0 +1,509 @@
+        these points represent calibration measurements taken at different times.
+        The function figures out which two calibration points are closest in time
+        to the specific measurement time you're interested in.
+
+    3. **Calculating Factors**: Once it finds these two nearby calibration points,
+        the function calculates a correction factor by drawing a straight line
+        between them (linear interpolation). This factor helps adjust the measurement
+        to make it more accurate, considering how the instrument's sensitivity changed
+        between those two calibration points.
+
+    4. **Returning the Correction Factor**: Finally, the function returns this
+        correction factor. You can then use this factor to adjust or calibrate your
+        measurements at the specific time you're interested in. This ensures that
+        your measurements are as accurate as possible, taking into account the
+        instrument's changing sensitivity over time.
+    """
+    # NOTE: waiting on fake calibration data to write this.
+    pass
+
+
+def apply_in_flight_calibration(data):
+    """Apply in flight calibration to full cycle data.
+
+    These factors are used to account for changes in gain with time.
+
+    They are derived from the weekly electron calibration data.
+
+    Parameters
+    ----------
+    data : _type_
+        _description_
+    """
+    # calculate calibration factor
+    # Apply to all data
+    pass
+
+
+def populate_full_cycle_data(
+    l1a_data: xr.Dataset, packet_index: int, esa_table_num: int
+):
+    """Populate full cycle data array using esa lookup table and l1a_data.
+
+    Parameters
+    ----------
+    l1a_data : xr.Dataset
+        L1a data with full cycle data only
+    packet_index : int
+        Index of current packet in the whole packet list.
+    esa_table_num : int
+        ESA lookup table number
+
+    Returns
+    -------
+    np.array
+        Array with full cycle data populated
+    """
+    esa_lookup_table = read_lookup_table(esa_table_num)
+
+    # If esa lookup table number is 0, then populate using esa lookup table data
+    # with information that esa step ramps up in even column and ramps down
+    # in odd column every six steps.
+    if esa_table_num == 0:
+        # create new full cycle data array
+        full_cycle_data = np.zeros((24, 30, 7))
+
+        # Initialize esa_step_number and column_index.
+        # esa_step_number goes from 0 to 719 range where
+        # 720 came from 24 x 30. full_cycle_data array has (24, 30)
+        # dimension.
+        esa_step_number = 0
+        # column_index goes from 0 to 29 range where
+        # 30 came from 30 column in full_cycle_data array
+        column_index = -1
+
+        # Go through four quarter cycle data packets
+        for index in range(4):
+            uncompressed_counts = l1a_data["SCIENCE_DATA"].data[packet_index + index]
```

Comment:
I think we decided on decompressed for consistency.
```suggestion
            decompressed_counts = l1a_data["SCIENCE_DATA"].data[packet_index + index]
```